source/common/runtime/runtime_impl.h

```diff
@@ -55,41 +56,72 @@ struct RuntimeStats {
 };
 
 /**
- * Implementation of Snapshot that reads from disk.
+ * Implementation of Snapshot whose source the vector of layers passed to the constructor.
  */
-class SnapshotImpl : public Snapshot,
-                     public ThreadLocal::ThreadLocalObject,
-                     Logger::Loggable<Logger::Id::runtime> {
+class SnapshotImpl : public Snapshot, public ThreadLocal::ThreadLocalObject {
 public:
-  SnapshotImpl(const std::string& root_path, const std::string& override_path, RuntimeStats& stats,
-               RandomGenerator& generator, Api::OsSysCalls& os_sys_calls);
+  SnapshotImpl(RandomGenerator& generator, RuntimeStats& stats,
+               const std::vector<OverrideLayerSharedPtr>& layers);
```

Comment:
nit: can we pass `std::vector<OverrideLayerSharedPtr>&&` and move. I think all callers are move capable.